test/remote/gateways/remote_rapyd_test.rb

```diff
@@ -205,4 +208,43 @@ def test_transcript_scrubbing
     assert_scrubbed(@gateway.options[:secret_key], transcript)
     assert_scrubbed(@gateway.options[:access_key], transcript)
   end
+
+  def test_successful_authorize_with_3ds_v1_options
+    options = {
+      pm_type: 'gb_visa_card',
+      curency: 'EUR'
+    }
+    options[:three_d_secure] = {
+      version: '1.0.2',
+      cavv: 'jJ81HADVRtXfCBATEp01CJUAAAA=',
+      xid: '00000000000000000501',
+      eci: '02',
+      ds_transaction_id: '97267598-FAE6-48F2-8083-C23433990FBC'
+    }
+
+    response = @gateway.authorize(105000, @credit_card, options)
+    assert_success response
+    assert_equal 'ACT', response.params['data']['status']
+    assert_equal '3d_verification', response.params['data']['payment_method_data']['next_action']
+    assert response.params['data']['redirect_url']
+  end
+
+  def test_successful_authorize_with_3ds_v2_options
+    options = {
+      pm_type: 'gb_visa_card',
+      curency: 'EUR'
+    }
+    options[:three_d_secure] = {
+      version: '2.1.0',
+      cavv: 'jJ81HADVRtXfCBATEp01CJUAAAA=',
+      xid: '00000000000000000501',
+      eci: '02'
+    }
+
+    response = @gateway.authorize(105000, @credit_card, options)
+    assert_success response
+    assert_equal 'ACT', response.params['data']['status']
+    assert_equal '3d_verification', response.params['data']['payment_method_data']['next_action']
+    assert response.params['data']['redirect_url']
+  end
```

Comment:
Should unit tests be added as well?

Reply:
I could add unit tests to verify that values are being correctly added to the body of the request.